api/app/auto_spatial_advisory/process_fuel_type_area.py

```diff
@@ -0,0 +1,225 @@
+""" Code relating to processing high HFI area per fire zone
+"""
+
+import logging
+import numpy as np
+import os
+import tempfile
+from datetime import date, datetime
+from osgeo import gdal
+from time import perf_counter
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.future import select
+from sqlalchemy.sql import text
+from app import config
+from app.auto_spatial_advisory.common import get_s3_key
+from app.auto_spatial_advisory.run_type import RunType
+from app.db.database import DB_READ_STRING, get_async_write_session_scope
+from app.db.models.auto_spatial_advisory import AdvisoryFuelStats, SFMSFuelType
+from app.db.crud.auto_spatial_advisory import get_all_hfi_thresholds, get_all_sfms_fuel_types, get_run_parameters_id, save_advisory_fuel_stats
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_warped_fuel_type_s3_key(bucket):
+    """
+    Returns the key to the fuel type layer that has been reprojected to the Lambert Conformal Conic spatial reference and
+    transformed to match the extent and spatial reference of hfi files output by sfms.
+    """
+
+    # The filename in our object store, prepended with "vsis3" - which tells GDAL to use
+    # it's S3 virtual file system driver to read the file.
+    # https://gdal.org/user/virtual_file_systems.html
+    key = f"/vsis3/{bucket}/ftl/fuel_types_lambert_2000.tif"
+    return key
+
+
+def classify_by_threshold(source_data: np.array, threshold: int):
+    """
+    Classifies the provided 2-d array based on the provided threshold. When the threshold is 1, all cells with an hfi value
+    in the range of 4k - 10k are assigned a value of 1 and other cells are assigned a value of 0. When the threshold is 2,
+    all cells with an hfi greater than 10k are assigned a value of 1 and other cells are assigned a value of 0.
+
+    :param source_data: A 2-d array with values representing fuel types.
+    :param threshold: The current threshold being processed, 1 = 4k-10k, 2 = > 10k.
+    """
+    if threshold == 1:
+        # advisory
+        classified = np.where(source_data < 4000, 0, source_data)
```

Comment:
Can we reuse the classify logic that already exists in `classify_hfi.py`?

Reply:
Unfortauntely not. That code creates three classes, 0, 1 and 2. For the raster multiplication in this ticket I need 0 and 1. The zero values 'remove' a fuel type cell in the multiplication process while a 1 retains the fuel type cell.